Kitodo/src/main/java/org/kitodo/production/interfaces/activemq/TaskActionProcessor.java

```diff
@@ -0,0 +1,219 @@
+import org.kitodo.data.database.enums.TaskStatus;
+import org.kitodo.data.database.exceptions.DAOException;
+import org.kitodo.data.elasticsearch.exceptions.CustomResponseException;
+import org.kitodo.data.exceptions.DataException;
+import org.kitodo.exceptions.ProcessorException;
+import org.kitodo.production.services.ServiceManager;
+import org.kitodo.production.services.data.TaskService;
+import org.kitodo.production.services.workflow.WorkflowControllerService;
+
+/**
+ * This is a web service interface to modify task status.
+ */
+public class TaskActionProcessor extends ActiveMQProcessor {
+
+    public static final String KEY_CORRECTION_TASK_ID = "correctionTaskId";
+    public static final String KEY_MESSAGE = "message";
+    public static final String KEY_TASK_ACTION = "action";
+    public static final String KEY_TASK_ID = "id";
+    private final TaskService taskService = ServiceManager.getTaskService();
+    private final WorkflowControllerService workflowControllerService;
+
+    /**
+     * The default constructor looks up the queue name to use in kitodo_config.properties. If that is not configured and
+     * “null” is passed to the super constructor, this will prevent ActiveMQDirector.registerListeners() from starting
+     * this service.
+     */
+    public TaskActionProcessor() {
+        super(ConfigCore.getOptionalString(ParameterCore.ACTIVE_MQ_TASK_ACTION_QUEUE).orElse(null));
+        workflowControllerService = new WorkflowControllerService();
+    }
+
+    /**
+     * This is the main routine processing incoming messages. It gets the task id and the task action for processing.
+     * Every action has its own behavior so please read the comment on the action for more information.
+     *
+     * @param mapMessageObjectReader
+     *         the incoming message
+     */
+    @Override
+    protected void process(MapMessageObjectReader mapMessageObjectReader) throws ProcessorException, JMSException {
+        Integer taskId = mapMessageObjectReader.getMandatoryInteger(KEY_TASK_ID);
+        String taskActionField = mapMessageObjectReader.getMandatoryString(KEY_TASK_ACTION);
+        TaskAction taskAction;
+        try {
+            taskAction = TaskAction.valueOf(mapMessageObjectReader.getMandatoryString(KEY_TASK_ACTION));
+        } catch (IllegalArgumentException e) {
+            throw new ProcessorException("Unknown task action " + taskActionField);
+        }
+
+        try {
+            Task currentTask = taskService.getById(taskId);
+            if (Objects.isNull(currentTask)) {
+                throw new ProcessorException("Task with id " + taskId + " not found.");
+            }
+
+            processAction(mapMessageObjectReader, taskAction, currentTask);
+
+            if (mapMessageObjectReader.hasField(KEY_CORRECTION_TASK_ID)) {
+                for (Task task : currentTask.getProcess().getTasks()) {
+                    taskService.saveToIndex(task, true);
```

Comment:
Saving all task to index of a process looks for me still like solving a hidden issue, even this is only called when a correction id is submitted.